hrms/hr/doctype/full_and_final_statement/full_and_final_statement.py

```diff
@@ -8,6 +8,15 @@
 
 
 class FullandFinalStatement(Document):
+	def on_change(self):
+		for payable in self.payables:
+			if payable.component == "Gratuity":
+				frappe.db.set_value(
+					"Gratuity",
+					payable.reference_document,
+					{"status": self.status, "paid_amount": payable.amount},
+				)
```

Comment:
set paid amount of gratuity doc and call set_status method so that the responsibility of setting the correct status is handled within gratuity itself

Also doesn't handle cancellation